lein-cljfmt/src/leiningen/cljfmt.clj

```diff
@@ -6,17 +6,17 @@
             [leiningen.core.main :as main]
             [leiningen.cljfmt.diff :as diff]))
 
-(defn clojure-file? [file]
-  (re-find #"\.clj[sx]?" (str file)))
+(defn clojure-file? [pattern file]
+  (re-find pattern (str file)))
 
-(defn clojure-files [dir]
-  (filter clojure-file? (file-seq (io/file dir))))
+(defn clojure-files [pattern dir]
```

Comment:
Now that you allow any pattern, the names `clojure-file?` and `clojure-files` don't really match the functionality. Instead, perhaps something like:

```clojure
(defn grep [re dir]
  (filter #(re-find re (str %)) (file-seq (io/file dir))))
```